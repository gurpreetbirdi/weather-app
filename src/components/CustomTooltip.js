import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Card, CardContent, Typography, CardActions } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import moment from 'moment';

import { constants } from '../config/constants';

const useStyles = makeStyles(() => ({
  alignment: { alignItems: 'center' },
  margin: { margin: '0.1em', padding: '0.1em' },
  zeroPadding: { padding: '0px' },
}));

const CustomTooltip = ({ active, payload = [], label }) => {
  const classes = useStyles();

  if (active) {
    return (
      <Card variant="outlined">
        <CardActions>
          <CardContent className={classes.zeroPadding}>
            <Grid container className={classes.alignment}>
              <Typography variant="h5" component="h2" className={classes.margin}>
                Temp:
              </Typography>
              <Typography variant="h6" component="h2" className={classes.margin} color="textSecondary">
                {label}
              </Typography>
            </Grid>
            <Grid container className={classes.alignment}>
              <Typography variant="h5" component="h2" className={classes.margin}>
                Time:
              </Typography>
              <Typography
                variant="h6"
                component="h2"
                className={classes.margin}
                style={{ whiteSpace: 'nowrap' }}
                color="textSecondary">
                {moment(payload[0]?.payload['dt_txt']).format(constants.TIME_FORMAT)}
              </Typography>
            </Grid>
          </CardContent>
        </CardActions>
      </Card>
    );
  }

  return null;
};

CustomTooltip.propTypes = {
  active: PropTypes.bool,
  label: PropTypes.string,
  payload: PropTypes.array,
};

export default CustomTooltip;
