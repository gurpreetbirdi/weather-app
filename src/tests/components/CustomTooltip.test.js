import React from 'react';
import { shallow } from 'enzyme';

import CustomTooltip from '../../components/CustomTooltip';
import { payload } from '../fixtures/payload';

const label = '53.98 Â°F';

test('should render CustomTooltip correctly', () => {
  const wrapper = shallow(<CustomTooltip label={label} payload={payload} active />);
  expect(wrapper).toMatchSnapshot();
});

test('should CustomTooltip return null if not active', () => {
  const wrapper = shallow(<CustomTooltip label={label} payload={payload} active={false} />);
  expect(wrapper).toMatchSnapshot();
});
