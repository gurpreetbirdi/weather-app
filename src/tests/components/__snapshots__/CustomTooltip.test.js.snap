// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should CustomTooltip return null if not active 1`] = `""`;

exports[`should render CustomTooltip correctly 1`] = `
<WithStyles(ForwardRef(Card))
  variant="outlined"
>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-zeroPadding-3"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-alignment-1"
        container={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-margin-2"
          component="h2"
          variant="h5"
        >
          Temp:
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-margin-2"
          color="textSecondary"
          component="h2"
          variant="h6"
        >
          53.98 Â°F
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-alignment-1"
        container={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-margin-2"
          component="h2"
          variant="h5"
        >
          Time:
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          className="makeStyles-margin-2"
          color="textSecondary"
          component="h2"
          style={
            Object {
              "whiteSpace": "nowrap",
            }
          }
          variant="h6"
        >
          09 PM
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
